lib/decorateWithMetrics.js

```diff
@@ -136,62 +138,91 @@ function toDuration(millis, raw = false) {
+    const {totalTime, summaries, lastCompleteCall} = getMetrics();
 
-    const prettyData = _.chain(summaries)
+    const tableSummary = MDTable.factory({
+        heading: ['method', 'total time spent', 'max', 'min', 'avg', 'time last call completed', 'calls made'],
+        leftAlignedCols: [0],
+    });
+
+    const methodCallTables = _.chain(summaries)
         .filter(method => method.avg > 0)
         .sortBy('avg')
         .reverse()
-        .map(({
-            calls, methodName, lastCall, ...summary
-        }) => {
-            const prettyTimes = _.chain(summary)
-                .map((value, key) => ([key, toDuration(value, raw)]))
-                .object()
-                .value();
-
-            const prettyCalls = calls.map(call => ({
-                startTime: toDuration(call.startTime, raw),
-                endTime: toDuration(call.endTime, raw),
-                duration: toDuration(call.duration, raw),
-                args: JSON.stringify(call.args)
-            }));
-
-            return {
+        .map(({methodName, calls, ...methodStats}) => {
+            tableSummary.addRow(
                 methodName,
-                ...prettyTimes,
-                'time last call completed': toDuration(lastCall.endTime, raw),
-                calls: calls.length,
-                prettyCalls,
-            };
+                toDuration(methodStats.total, raw),
+                toDuration(methodStats.max, raw),
+                toDuration(methodStats.min, raw),
+                toDuration(methodStats.avg, raw),
+                toDuration(methodStats.lastCall.endTime, raw),
+                calls.length,
+            );
+
+            const callsTable = MDTable.factory({
+                title: methodName,
+                heading: ['start time', 'end time', 'duration', 'args'],
+                leftAlignedCols: [3],
+                rows: calls.map(call => ([
+                    toDuration(call.startTime, raw),
+                    toDuration(call.endTime, raw),
+                    toDuration(call.duration, raw),
+                    call.args.map(String).join(', ').slice(0, 60), // Restrict cell width to 60 chars max
+                ]))
+            });
+
+            return callsTable;
         })
         .value();
 
-    /* eslint-disable no-console */
-    console.group('Onyx Benchmark');
-    console.info('  Total: ', toDuration(totalTime, raw));
-    console.info('  Last call finished at: ', toDuration(lastCompleteCall.endTime, raw));
+    if (/csv|json|string/i.test(format)) {
+        const allTables = [tableSummary, ...methodCallTables];
+
+        return allTables.map((table) => {
+            switch (format.toLowerCase()) {
+                case 'csv':
+                    return table.toCSV();
+                case 'json':
+                    return table.toJSON();
+                default:
+                    return table.toString();
+            }
+        }).join('\n\n');
```

Comment:
Any specific reason we join with 2 new lines?

Reply:
To distinct separate tables easier